sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -105,6 +105,11 @@ abstract class Expression extends TreeNode[Expression] {
       val isNull = ctx.freshName("isNull")
       val value = ctx.freshName("value")
       val eval = doGenCode(ctx, ExprCode("", isNull, value))
+
+      // Records current input row and variables of this expression.
+      eval.inputRow = ctx.INPUT_ROW
+      eval.inputVars = findInputVars(ctx, eval)
```

Comment:
shall we do one more thing here?
```
eval.isNull = if (this.nullable) eval.isNull else "false"
```

Reply:
actually we can be more aggresive and do `val isNull = if (this.nullable) eval.isNull else "false"`, and see if there is any compile errors.

Reply:
Ok. Let me try it in next commit.

Reply:
The aggressive version (`val isNull = if (this.nullable) ctx.freshName("isNull") else "false"`) fails compilation when testing locally.

Reply:
`eval.isNull = if (this.nullable) eval.isNull else "false"` seems ok.